src/VisualStudio/Core/Def/ProjectSystem/CPS/IWorkspaceProjectContext.cs

```diff
@@ -72,6 +73,7 @@ internal interface IWorkspaceProjectContext : IDisposable
         void RemoveAnalyzerConfigFile(string filePath);
 
         void StartBatch();
+        IAsyncDisposable CreateBatchScope();
```

Comment:
ooh, would be nice to use this for all hosts.  considering deprecating StartBatch and EndBatchAsync.

Reply:
Yeah, this API is cleaner. It's actually how it's implemented under the covers. I won't track a specific deprecation plan for these though since we'll just be moving away from this interface entirely.